umm-spec-lib/src/cmr/umm_spec/migration/version/core.clj

```diff
@@ -27,24 +27,36 @@
 
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;;; Utility Functions
+(def ^:private values-are-equal 0)
+(def ^:private begin-less-than-end -1)
+(def ^:private begin-greater-than-end 1)
 
 (defn- customized-compare
   "Customizing the compare because normal string compare results in
    1.n > 1.10 and doesn't convert the version_steps correctly(1< n <10).
-   Assuming the version only contains two parts part1.part2.
-   both part1 and part2 are integers."
+   The inputs assume the version only contains integers separated by dots.
+   This scheme allows version numbers like 1.10.2 and 1.10.3.25 etc."
   [begin end]
-  (let [begin-parts (string/split begin #"\.")
-        end-parts (string/split end #"\.")
-        begin-part1 (read-string (first begin-parts))
-        begin-part2 (read-string (last begin-parts))
-        end-part1 (read-string (first end-parts))
-        end-part2 (read-string (last end-parts))
-        part1-compare (- begin-part1 end-part1)
-        part2-compare (- begin-part2 end-part2)]
-    (if (zero? part1-compare)
-      part2-compare
-      part1-compare)))
+  (let [begin-tmp (map #(Integer/parseInt %) (clojure.string/split begin #"\."))
+        end-tmp (map #(Integer/parseInt %) (clojure.string/split end #"\."))]
+    (loop [begin begin-tmp
+           end end-tmp]
+      (println begin end)
```

Comment:
Remove println or change it to use the logging framework.

Reply:
I'm still parsing through this, will update this comment when I figure out a better way to do this.

Reply:
Thanks

Reply:
Fixed